src/parselmouth/PointProcess.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2019-2021  Yannick Jadoul
+ *
+ * This file is part of Parselmouth.
+ *
+ * Parselmouth is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Parselmouth is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Parselmouth.  If not, see <http://www.gnu.org/licenses/>
+ */
+
+#include "Parselmouth.h"
+
+#include "TimeClassAspects.h"
+
+#include <praat/fon/PointProcess.h>
+#include <praat/fon/Pitch_to_PointProcess.h>
+#include <praat/fon/VoiceAnalysis.h>
+
+#include <vector>
+#include <tuple>
+
+#include <pybind11/stl.h>
+
+namespace py = pybind11;
+using namespace py::literals;
+using namespace std::string_literals;
+
+namespace parselmouth {
+
+PRAAT_CLASS_BINDING(PointProcess) {
+	addTimeFunctionMixin(*this);
+
+	// FORM(NEW1_PointProcess_createEmpty, U"Create an empty PointProcess", U"Create empty PointProcess...")
+	def(py::init([](double startTime, double endTime, integer initialMaxnt) {
```

Comment:
`initialMaxnt` doesn't seem to be part of `"Create an empty PointProcess"`? Do we really want to expose the size of the internally allocated buffer to Python users?

Wouldn't it be more useful to provide a constructor from an array of times, or so?

Reply:
Thanks for checking what `initialMaxnt` is (I grossed over it). If that is what it is, I suggest dropping that argument and create another constructor for `PointProcess_createPoissonProcess`.

Also, a big yes for passing time array in. That would be very convenient (if you ever want to create it yourself). I can do that

Reply:
:+1: Great! My criteria tends to be to use the arguments passed to the Praat `FORM`s, but I've made errors and mistakes against that as well. (And of course, for good reason, we can diverge from that.)

`PointProcess_createPoissonProcess` makes sense, btw, though it probably needs to be a named kind of constructor, to avoid confusion with the "normal" constructor? `PointProcess.create_poisson_process()` (or a better name, if you find one) ?

> Also, a big yes for passing time array in. That would be very convenient (if you ever want to create it yourself). I can do that

Great! No rush on having that in immediately, though. Just a `// TODO` also works, for now. Though it is good fun to try doing. I suggest looking at the `Sound` constructor to get this.

Reply:
Or just accept a `set`, of course, since the order doesn't matter? It all depends a bit on how what makes most sense as Python interface. But just an array might work (see `PointProcess_addPoints`, which also seems to sort points afterwards). But since this doesn't really exist as Praat interface, there's more freedom to pick the "best" Python API :-)